s3fs/mapping.py

```diff
@@ -65,24 +63,16 @@ def __setitem__(self, key, value):
             f.write(value)
 
     def keys(self):
-        return iter(map(lambda x: x[len(self.root) + 1:], self.s3.walk(self.root)))
+        return (x[len(self.root) + 1:] for x in self.s3.walk(self.root))
```

Comment:
Generators are Iterators, not Iterables.  

Looking at the travis logs (available if you click the red X) it was an import issue.

```
$ py.test -x --doctest-modules --pyargs s3fs
============================= test session starts ==============================
platform linux -- Python 3.4.4, pytest-2.9.1, py-1.4.31, pluggy-0.3.1
rootdir: /home/travis/build/dask/s3fs, inifile: 
collecting 8 items / 1 errors
==================================== ERRORS ====================================
_________________ ERROR collecting s3fs/tests/test_mapping.py __________________
s3fs/tests/test_mapping.py:2: in <module>
    from s3fs.mapwrap import S3Map
E   ImportError: No module named 's3fs.mapwrap'
!!!!!!!!!!!!!!!!!!!! Interrupted: stopping after 1 failures !!!!!!!!!!!!!!!!!!!!
```